index.js

```diff
@@ -97,6 +97,23 @@ function getCircularValueOption (options) {
   return '"[Circular]"'
 }
 
+function getDeterministicOption (options, key) {
+  let value
+  if (hasOwnProperty.call(options, key)) {
+    value = options[key]
+    if (typeof value === 'boolean') {
+      return value
+    }
+    if (typeof value === 'function' && value.length === 2) {
```

Comment:
I guess it should be fine to verify that the input is a function. `function comparator(...args) {}` would for example have a length of zero and that is totally legit to use like that.

```suggestion
    if (typeof value === 'function') {
```